framework/src/ble_manager/ble_manager_state.c

```diff
@@ -0,0 +1,336 @@
+		trble_attr_handle attr_handle = *(trble_attr_handle *)param->param[0];
+		trble_data *data = (trble_data *)param->param[1];
+
+		ret = ble_drv_attr_set_data(attr_handle, data);
+	}
+	case BLE_EVT_ATTR_GET_DATA_CMD: {
+		BLE_STATE_CHECK;
+
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		trble_attr_handle attr_handle = *(trble_attr_handle *)param->param[0];
+		trble_data *data = (trble_data *)param->param[1];
+
+		ret = ble_drv_attr_get_data(attr_handle, data);
+	}
+	case BLE_EVT_ATTR_REJECT_CMD: {
+		BLE_STATE_CHECK;
+
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		trble_attr_handle attr_handle = *(trble_attr_handle *)param->param[0];
+		uint8_t app_errorcode = *(uint8_t *)param->param[1];
+
+		ret = ble_drv_attr_reject(attr_handle, app_errorcode);
+	}
+	case BLE_EVT_GET_MAC_BY_CONN_CMD: {
+		BLE_STATE_CHECK;
+
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		trble_conn_handle con_handle = *(trble_conn_handle *)param->param[0];
+		uint8_t *bd_addr = (uint8_t *)param->param[1];
+
+		ret = ble_drv_get_mac_addr_by_conn_handle(con_handle, bd_addr);
+	}
+	case BLE_EVT_GET_CONN_BY_MAC_CMD: {
+		BLE_STATE_CHECK;
+
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		uint8_t *bd_addr = (uint8_t *)param->param[0];
+		trble_conn_handle *con_handle = (trble_conn_handle *)param->param[1];
+
+		ret = ble_drv_get_conn_handle_by_addr(bd_addr, con_handle);
+	}
+	case BLE_EVT_SET_ADV_DATA_CMD: {
+		BLE_STATE_CHECK;
+
+		trble_data *data = (trble_data *)msg->param;
+		ret = ble_drv_set_adv_data(data);
+	}
+	case BLE_EVT_SET_ADV_RESP_CMD: {
+		BLE_STATE_CHECK;
+
+		trble_data *data = (trble_data *)msg->param;
+		ret = ble_drv_set_adv_resp(data);
+	}
+	case BLE_EVT_CONN_IS_ACTIVE_CMD: {
+		BLE_STATE_CHECK;
+
+		bool *is_active = (bool *)msg->param;
+		ret = ble_drv_conn_is_active(is_active);
+	}
+	case BLE_EVT_CONN_IS_ANY_ACTIVE_CMD: {
+		BLE_STATE_CHECK;
+
+		bool *is_active = (bool *)msg->param;
+		ret = ble_drv_conn_is_any_active(is_active);
+	}
+	case BLE_EVT_GET_BONDED_DEV_CMD: {
+		BLE_STATE_CHECK;
+
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		trble_bonded_device_list_s *list = (trble_bonded_device_list_s *)param->param[0];
+		uint16_t *device_count = (uint16_t *)param->param[1];
+
+		ret = ble_drv_get_bonded_device(list, device_count);
+	}
+	case BLE_EVT_START_ADV_CMD: {
+		BLE_STATE_CHECK;
+
+		ret = ble_drv_start_adv();
+	}
+	case BLE_EVT_START_ADV_DIRECTED_CMD: {
+		BLE_STATE_CHECK;
+
+		uint8_t *bd_addr = (uint8_t *)msg->param;
+		ret = ble_drv_start_adv_directed(bd_addr);
+	}
+	case BLE_EVT_STOP_ADV_CMD: {
+		BLE_STATE_CHECK;
+
+		ret = ble_drv_stop_adv();
+	}
+	case BLE_EVT_SET_ADV_INTERVAL_CMD: {
+		BLE_STATE_CHECK;
+
+		uint16_t interval = *(uint16_t *)msg->param;
+		ret = ble_drv_set_adv_interval(interval);
+	}
+	default:
+		break;
+	}
+
+handle_req_done:
+	BLE_LOG_INFO("[BLE] T%d <-- _handle_request\n", getpid());
```

Comment:
For using `getpid()`, please include `unistd.h`.

Reply:
ok, I will add it